app/Http/Controllers/ImageController.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use App\Http\Requests\CategoryRequest;
+
+class ImageController extends Controller
+{
+    public function uploadIcon(CategoryRequest $request)
+    {
+
+        if ($request->hasFile('icon')) {
+            $file = $request->file('icon');
+            $path = $file->store('icons', 'public');
+
+
+            return response()->json(['path' => $path], 200);
```

Comment:
You uploaded icon but we don't know which category it refers to, you didn't save the path in the category record. Check route model binding in the documentation. 
Also, you applied CategoryRequest so you will get 422 error if you don't provide required params.